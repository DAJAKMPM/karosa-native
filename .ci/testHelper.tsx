import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
