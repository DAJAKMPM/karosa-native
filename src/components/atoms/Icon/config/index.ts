/**
 *
 * @format
 *
 */

export const ZERO = 0;
