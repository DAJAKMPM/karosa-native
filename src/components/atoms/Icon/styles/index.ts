/**
 *
 * @format
 *
 */

import {StyleSheet} from 'react-native';

const IconStyles = StyleSheet.create({
});

export default IconStyles;
