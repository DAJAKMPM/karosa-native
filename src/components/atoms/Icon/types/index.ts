/**
 *
 * @format
 *
 */

import { StyleProp, ViewStyle } from "react-native";

export type PropsType = {
  group: string;
  name: string;
  extraStyle: StyleProp<ViewStyle>;
  height: number;
  width: number;
};
