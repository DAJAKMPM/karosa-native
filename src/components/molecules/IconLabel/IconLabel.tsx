/**
 *
 * IconLabel
 * @format
 *
 */

import React, { FC } from "react";
import { View } from "react-native";
import Text from "@app/atoms/Text";

import type { PropsType } from "./types";
import IconLabelStyles from "./styles";

const IconLabel: FC<PropsType> = (props) => {
  const { icon, title, subtitle, containerStyle } = props;

  return (
    <View style={[IconLabelStyles.container, containerStyle]}>
      {icon}
      <View style={IconLabelStyles.detailsContainer}>
        <Text text={title} textStyle={IconLabelStyles.title} />
        <Text text={subtitle} textStyle={IconLabelStyles.subtitle} />
      </View>
    </View>
  );
};

export default IconLabel;
