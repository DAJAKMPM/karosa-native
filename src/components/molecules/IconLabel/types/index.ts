/**
 *
 * @format
 *
 */

import { ReactElement } from "react";
import { StyleProp, ViewStyle } from "react-native";

export type PropsType = {
  icon: ReactElement;
  title: string;
  subtitle: string;
  containerStyle?: StyleProp<ViewStyle>;
};
