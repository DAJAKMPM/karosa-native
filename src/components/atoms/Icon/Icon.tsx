/**
 *
 * Icon
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import Icons from "@app/assets/icons";

import type { PropsType } from "./types";
import { ZERO } from "./config";

const Icon: FC<PropsType> = (props) => {
  const { group, name, extraStyle, height, width } = props;

  const icon = Icons[group](
    name,
    extraStyle,
    width ? width : ZERO,
    height ? height : ZERO
  );

  return <Fragment>{icon}</Fragment>;
};

export default Icon;
