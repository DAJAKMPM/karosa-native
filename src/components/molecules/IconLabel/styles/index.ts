/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme, SPACING } from "@app/styles";

const IconLabelStyles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row",
    backgroundColor: theme.colors.white,
  },
  detailsContainer: {
    flex: 1,
    marginLeft: SPACING.lg,
    justifyContent: "center",
  },
  title: {
    ...theme.textRegular,
    fontWeight: "400",
  },
  subtitle: { ...theme.textRegular, fontWeight: "300" },
});

export default IconLabelStyles;
