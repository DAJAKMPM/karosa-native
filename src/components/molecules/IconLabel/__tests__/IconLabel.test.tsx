import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Icon from "@app/atoms/Icon";
import IconLabel from "../IconLabel";

describe("IconLabel Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <IconLabel
        icon={<Icon group="login" name="phone" width={20} height={20} />}
        subtitle="test"
        title="test"
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
