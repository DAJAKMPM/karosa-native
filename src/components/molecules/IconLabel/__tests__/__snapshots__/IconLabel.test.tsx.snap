// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLabel Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View>
      <SvgMock
        height={20}
        width={20}
      />
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 12,
          "marginTop": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 18.75,
            "marginBottom": 12,
          }
        }
      >
        test
      </Text>
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 16,
            "fontWeight": "300",
            "lineHeight": 18.75,
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;
